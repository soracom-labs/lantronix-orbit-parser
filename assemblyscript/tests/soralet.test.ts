import { join } from "path";
import { Runtime } from "./runtime";
import { inputs } from "./inputs";
import { writeFileSync } from "fs";
import { SoraletPayload } from "./soralet_payload";

describe("soralet test suite", (): void => {
  let num = 0;
  for (const input of inputs) {
    describe("Environment", () => {
      test("input has expected props", () => {
        expect(input).toHaveProperty("body");
        expect(input).toHaveProperty("direction");
        expect(input).toHaveProperty("contentType");
      });
    });

    writeFileSync(
      join(__dirname, "inputs", `${++num}.json`),
      JSON.stringify(toSoracomCliCompatibleJson(input), null, 2),
    );

    const runtime = new Runtime(join(__dirname, "..", "build", "soralet.wasm"), input);
    const result = runtime.exec();

    describe("WASM", () => {
      test("result code should be 0", () => {
        expect(result.resultCode).toBe(0);
      });

      test("content type should be application/json", () => {
        expect(result.contentType).toBe("application/json");
      });

      test("result should be expected", () => {
        expect(result.body).toMatchSnapshot();
      });
    });
  }
});

// rename `body` to `payload` to compatible with `soracom soralets exec` command's `body` parameter
function toSoracomCliCompatibleJson(input: SoraletPayload): any {
  const j = JSON.parse(JSON.stringify(input));
  j.payload = j.body;
  delete j.body;
  return j;
}
