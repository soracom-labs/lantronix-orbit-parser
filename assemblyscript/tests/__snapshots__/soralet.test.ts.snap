// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`soralet test suite WASM result should be expected 1`] = `
Object {
  "gprmc": Object {
    "latitude": 35.705833,
    "longitude": 139.7775,
    "v0": "GPRMC",
    "v1": "071054.000",
    "v10": "",
    "v11": "*10 ",
    "v2": "V",
    "v3": "3542.3459",
    "v4": "N",
    "v5": "13946.6537",
    "v6": "E",
    "v7": "",
    "v8": "",
    "v9": "250122",
  },
  "obdPos": Object {
    "speed": 0,
  },
}
`;

exports[`soralet test suite WASM result should be expected 2`] = `
Object {
  "gprmc": Object {
    "latitude": 35.705833,
    "longitude": 139.7775,
    "v0": "GPRMC",
    "v1": "073545.000",
    "v10": "",
    "v11": "*17 ",
    "v2": "V",
    "v3": "3542.3459",
    "v4": "N",
    "v5": "13946.6537",
    "v6": "E",
    "v7": "",
    "v8": "",
    "v9": "250122",
  },
  "obdEvent": Object {
    "batt": 3.695,
    "ignition": 1,
    "power": 12.07,
    "temp": 30,
  },
  "obdPos": Object {
    "speed": 0,
  },
}
`;
